sycl/include/CL/sycl/queue.hpp

```diff
@@ -305,9 +418,14 @@ class queue {
   }
 
 private:
-  std::shared_ptr<detail::queue_impl> impl;
+  shared_ptr_class<detail::queue_impl> impl;
   template <class Obj>
   friend decltype(Obj::impl) detail::getSyclObjImpl(const Obj &SyclObject);
+
+  /// A template-free version of submit.
+  event submit_impl(function_class<void(handler &)> CGH);
```

Comment:
Are we not removing T in submit because it's in the spec? 
What failure will we see in runtime if T passed to submit is not function_class?

Reply:
@garimagu table 4.22 of the spec explicitly specifies what T may be. Those are all callable objects of the same signature. So, user is not expected to pass anything incompatible with function_class and they get a compilation failure if the do. Furthermore, there're no behavior changes, even without this patch custom object causes failure.